File: single-number-II/singleNumber2.go
func singleNumber(nums []int) int {
    var a,b int = 0,0
    for _,c := range nums {
        var tmp int = (^a&b&c)|(a&^b&^c)
        b = (b^c)&^a
        a = tmp
    }
    return b
}

File: single-number-II/singleNumber2.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        b = 0
        for c in nums:
            ta=(~a&b&c)|(a&~b&~c);
            b=(~a&~b&c)|(~a&b&~c);
            a = ta
        return b
